### PR TITLE
Ci/update coverage action

### PR DESCRIPTION
**Pull request by nateglims**
Previous coverage action was deprecated.

It also did not analyze branch coverage.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -35,11 +35,21 @@ jobs:
...
+          LINE_COVERAGE=$(lcov --list build/coverage.info | tail -n 1 | cut -d '|' -f 2 | sed -n "s/\([^%]*\)%.*/\1/p")
+          BRANCH_COVERAGE=$(lcov --rc lcov_branch_coverage=1 --list build/coverage.info | tail -n 1 | cut -d '|' -f 4 | sed -n "s/\([^%]*\)%.*/\1/p")
+          RESULT=0
+          echo "Line coverage:   $LINE_COVERAGE"
+          if [[ $(echo "$LINE_COVERAGE < 100" | bc) -ne 0 ]]; then
+            echo "Line Coverage is too low."
+            RESULT=1
+          fi
+          echo "Branch coverage: $BRANCH_COVERAGE"
+          if [[ $(echo "$BRANCH_COVERAGE < 100" | bc) -ne 0 ]]; then
```
Comment (yourslab) [inline .github/workflows/ci.yml:48]: Can an `env` parameter be used to specify a minimum?
https://github.com/FreeRTOS/coreMQTT/pull/93/commits/c68fff779c44878c58707f9773a3ab699477cd2d#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR32

Comment (nateglims) [reply .github/workflows/ci.yml]: Yes, that works. Thanks for the suggestion.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -35,11 +35,23 @@ jobs:
           lcov --rc lcov_branch_coverage=1 --remove build/coverage.info '*mocks*' --output-file build/coverage.info
           lcov --rc lcov_branch_coverage=1 --list build/coverage.info
       - name: Check Coverage
-        uses: ChicagoFlutter/lcov-cop@v1.0.2
-        with:
-          path: "build/coverage.info"
-          min_coverage: 100
-          exclude: "**/*test*"
+        env:
+          MIN_COVERAGE: 100
```
Comment (aggarw13) [inline .github/workflows/ci.yml:39]: Can you print the minimum coverage value as 
`echo "Required line and branch coverages:    $MIN_COVERAGE"`

Comment (nateglims) [reply .github/workflows/ci.yml:39]: ok

**Review by aggarw13 — APPROVED** [review]

**Review by yourslab — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**